addon/mixins/scoped-query-cache.js

```diff
@@ -73,5 +76,21 @@ export default Ember.Mixin.create({
 
       return results;
     });
+  },
+
+  scheduleCacheDecay(delay = CACHE_DECAY_TIMEOUT, callback) {
+    this.decayId = setTimeout(() => {
+      this.cache = this._createCache();
+
+      callback && typeof callback === 'function' && callback();
```

Comment:
So I don't think we want to do that. That makes assumptions about the behavior, and couples it to the API. While it would be convenient to do that, we don't want to necessarily be prescriptive about how people use this functionality.

Additionally, this is a mixin, therefore it has no notion of `deactivate` - that's a `Route` thing. We should allow routes to bind that behavior. We should, however, be binding the cache flush in `destroy`, which would be generic across all object types.

My thought is that Routes will implement `deactivate`, which would schedule the decay. 

Does this make sense?